jdaviz/configs/cubeviz/plugins/viewers.py

```diff
@@ -144,46 +142,62 @@ def _get_marks_for_layers(self, layers):
 
     def _on_subset_delete(self, msg):
         # delete any ShadowSpatialSpectral mark for which either of the spectral or spatial marks
-        # no longer exists
-        spectral_marks = self._get_marks_for_layers(self._get_spectral_subset_layers())
-        spatial_marks = self._get_marks_for_layers(self._get_spatial_subset_layers())
+        # no longer exists by matching the uuid of the msg subset to the uuid of the subsets
+        # in ShadowSpatialSpectral
+        spectral_marks = self._get_marks_for_layers([layer for layer in self._get_spectral_subset_layers() # noqa
+                                                     if layer.layer.uuid == msg.subset.uuid])
+        spatial_marks = self._get_marks_for_layers([layer for layer in self._get_spatial_subset_layers() # noqa
+                                                    if layer.layer.uuid == msg.subset.uuid])
```

Comment:
what is this change intended to fix/change?  This function should be triggered when `msg.subset` is deleted, so if anything I would expect this to be `!=` instead of `==`... but I didn't think that check was necessary at all since this should be called after the subset is deleted and `_get_*_subset_layers` should avoid the deleted layer.

Reply:
after offline discussion, it looks like these lines are not needed anymore and the changes to lines 152-153 should work on their own (and make sense).